### DISCUSSION TITLE
Can not connect to certain device on Linux (device not found, but scanning does find it)

### DISCUSSION THREAD

**Discussion by MarkusPiotrowski** [hbldh/bleak#1658] [Q&A]
I'm doing the same on Windows and Linux with two different BLE peripherals (Bleak version 0.22.2):

1. Using BleakScanner with `find_device_by_name` to get the `BLEDevice` object
2. Using BleakClient to connect to the device, using the `BLEDevice` object obtained from 1. 
3. I run the code in a Python virtual environment, on Linux in the `/home/` directory

A. This works fine for both devices when running the code on Windows (Win 10)
B. This works fine for one device when running the code on Linux (Ubuntu 22.04), but the other fails with a "device not found" exception. This also fails when using the MAC address instead of the `BLEDevice` object.

I wonder if I'm a victim of this Linux 'feature' that I don't understand:
> D-Bus Authentication
Connecting to the host DBus from within a user namespace will fail. This is because the remapped UID will not match the UID that the hosts sees. To work around this, you can hardcode a UID with the BLEAK_DBUS_AUTH_UID environment variable.

If so, what would I need to do exactly in my Python code to work around this?

**Comment by dlech** [comment]
> I wonder if I'm a victim of this Linux 'feature' that I don't understand:

No, this only applies to things like Docker containers, not Python virtual envs.



> If so, what would I need to do exactly in my Python code to work around this?

Not sure without seeing logs. Logging Bluetooth packets is the best way to see what is going on behind the scenes.

**Comment by MarkusPiotrowski** [comment]
Logging with Bleak's logging feature or Wireshark records? I did both without seeing an error condition (though I don't speak "bluetooth"), but am happy to provide the logs, if you may want to look at them.

**Reply by dlech** [reply]
Wireshark is usually the most helpful. Although in this case, if the device is disappearing from BlueZ, the Bleak debug logs could be helpful too.
